python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -1792,9 +1797,22 @@ func node_store_get{poseidon_ptr: PoseidonBuiltin*, node_store: NodeStore}(
     tempvar node_store = NodeStore(
         new NodeStoreStruct(node_store.value.dict_ptr_start, new_dict_ptr)
     );
-    // Cast the result to an OptionalInternalNode and return
-    tempvar res = OptionalInternalNode(cast(pointer, InternalNodeEnum*));
-    return res;
+
+    // Cast the result to a Bytes, hash it to check invariant and RLP-decode it.
+    tempvar encoded_node = Bytes(cast(pointer, BytesStruct*));
+    if (encoded_node.value.len == 0) {
+        let res = OptionalInternalNode(cast(0, InternalNodeEnum*));
+        return res;
+    }
```

Comment:
no hash to do in this case? in this case it means that the RLP encoding of the node is `b""`: in which cases does this happen?

Reply:
ah ok, I think this should be done prior to casting the pointer then:

```suggestion
    if (pointer == 0) {
        let res = OptionalInternalNode(cast(0, InternalNodeEnum*));
        return res;
    }
```

Reply:
and giving a second thought, when are we querying nodes that are not in the node store?